clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -76,14 +82,22 @@ object GarbageCollector {
       apiConf: APIConfigurations,
       hcValues: Broadcast[ConfMap]
   ): Set[(String, Array[Byte], Array[Byte])] = {
-    val location = ApiClient
-      .get(apiConf)
-      .getMetaRangeURL(repo, commitID)
+    val conf = configurationFromValues(hcValues)
+    val apiClient = ApiClient.get(apiConf)
+    val commit = apiClient.getCommit(repo, commitID)
+    if (StringUtils.isNotBlank(conf.get(LAKEFS_CONF_DEBUG_GC_MAX_COMMIT_DATE_KEY))) {
+      val maxDate = LocalDate.parse(conf.get(LAKEFS_CONF_DEBUG_GC_MAX_COMMIT_DATE_KEY), DateTimeFormatter.ISO_DATE)
+      val maxDateEpoch = maxDate.atStartOfDay(ZoneId.of("UTC")).toInstant.toEpochMilli() / 1000      
```

Comment:
This performs the conversion at every call to getRangeTuples.  I worry that it behave unpleasantly if I give it a bad date format.  Can we instead parse _once_ at startup, and then package up a time or millis-since-epoch or whatever into the executors?

Reply:
Done - passing it as configuration